### PR TITLE
Re-enable logging validation on Python 3.

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3564

### CODE REVIEW

**Review by exarkun — APPROVED** [review]
Thanks.  Looks good to me.  If it passes CI, please merge.

**Comment by itamarst** [comment]
I'm still seeing JSON encoding errors due to bytes in logs once I start running more tests on Python 3, so ... something is missing. Will try to see if I can figure it out before merging, otherwise there might be follow-up.

**Comment by itamarst** [comment]
OK, fixed. Trivial test-only change, so will merge when tests pass.

**Pull request merged**
